src/utils/resize-observer-wrapper.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+// This implements a wrapper to the ResizeObserver API, so that only one copy of
+// a ResizeObserver exists. This is much more performant than having one
+// ResizeObserver for each use.
+// This was inspired by the code in https://github.com/jaredLunde/react-hook/blob/master/packages/resize-observer/src/index.tsx
+
+function createResizeObserverWrapper() {
+  const callbacks = new Map();
```

Comment:
nit: it would be nice to write the type of this map to the definition, so it would be easier to see.